### PR TITLE
Use oc-select to select roles

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Bring oc-select component from ODS to people and links sidebar

## Motivation and Context
Enable keyboard navigation out of the box

## How Has This Been Tested?

- test environment: E2E

## Screenshots (if appropriate):
Screenshot 2021-04-08 at 13 52 41
Screenshot 2021-04-08 at 13 52 58

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsEdgeCases</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootEdgeCases</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFilePermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissionMultipleUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFilePermissionMultipleUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicDifferentRoles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderAdvPermissionsGrp</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderAdvancedPermissionMU</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>

































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionsUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>

























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>ResharingToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>







































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Resharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>







































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>







































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14447/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicDifferentRoles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14452/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14454/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14612/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>











</p></details>

**Comment by pascalwengerter** [comment]
@LukasHirt did you already rebase? 😇  CI seems to be pretty unhappy still, haha

**Comment by LukasHirt** [comment]
@pascalwengerter yep, rebase done. Already fixing the failing scenarios...

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14964/

<p>













</p></details>

**Comment by LukasHirt** [comment]
So apparently these two strings are not equal and are failing tests... 🤮 

image

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14971/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14971/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14971/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14971/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14974/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14974/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRename</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14974/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14974/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsersRoot2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14984/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14984/

<p>







</p></details>

**Comment by kulmann** [comment]
I get `Cannot set the requested share permissions for New folder` error messages upon creating links or shares with an oc10 backend. How is it for you? 🤔

**Comment by LukasHirt** [comment]
> I get Cannot set the requested share permissions for New folder error messages

No error for me 🤷

Screenshot 2021-04-23 at 9 14 37
Screenshot 2021-04-23 at 9 15 09

File: packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue [hunk 1]
```diff
@@ -1,17 +1,28 @@
 <template>
   <oc-grid gutter="small" child-width="1-1">
-    <roles-select
-      mode="collaborators"
-      :roles="roles"
-      :selected-role="role"
-      @roleSelected="selectRole"
-    />
+    <label tag="label" class="oc-label">
```
Comment (pascalwengerter) [inline packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue:3]: Is `<label tag="label"` valid or duplicate information? (Like <progress role="progress" or <nav role="navigation")

Comment (LukasHirt) [reply packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue:3]: Ah, I had originally `<translate tag="label"...` and forgot to remove that after switching back to the label. Will remove it

Comment (pascalwengerter) [reply packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue:3]: Happens someplace else in this PR btw :)

File: packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
   <oc-grid gutter="small" child-width="1-1">
-    <roles-select
-      mode="collaborators"
-      :roles="roles"
-      :selected-role="role"
-      @roleSelected="selectRole"
-    />
+    <label class="oc-label">
+      {{ $gettext('Role') }}
+      <oc-select
```
Comment (kulmann) [inline packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue:5]: Can you move the `oc-select` out of the label and use `for="files-collaborators-role-button"` on the label instead? Don't know if this is strictly necessary for a11y, but we have it like that now everywhere else, that label and input are in parallel in the DOM, not wrapped anymore.

Comment (LukasHirt) [reply packages/web-app-files/src/components/Collaborators/CollaboratorsEditOptions.vue:5]: > Don't know if this is strictly necessary for a11y

It should be valid as I know this syntax actually from Marcus :grin: But sure, I don't want to break any consistency...

**Comment by kulmann** [comment]
> > I get Cannot set the requested share permissions for New folder error messages
> 
> No error for me 🤷

Weird, not failing anymore. Probably didn't have a clean state. Sorry for the noise

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15014/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicDifferentRoles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15014/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15014/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15017/

<p>







</p></details>

**Comment by LukasHirt** [comment]
@kulmann @pascalwengerter Ready for re-review

**Pull request merged**